### PR TITLE
Refactor plugin fixtures to CJS

### PR DESCRIPTION
**Pull request by ybiquitous**
> Which issue, if any, is this issue related to?

Ref #5291

> Is there anything in the PR that needs further explanation?

Currently, we support plugins only for CJS. So, this change explicitly appends the `.cjs` extension.

### CODE REVIEW

File: lib/__tests__/fixtures/config-relative-plugin.json
Comment (Mouvedia) [inline lib/__tests__/fixtures/config-relative-plugin.json:1]: Did you miss https://github.com/stylelint/stylelint/blob/v16/lib/__tests__/fixtures/config-relative-plugin-nested.json#L2?

Comment (ybiquitous) [reply lib/__tests__/fixtures/config-relative-plugin.json:1]: No need to change the file because `plugin-nested-warn-about-foo` doesn't exist.

**Review by mattxwang — APPROVED** [review]
Overall, this LGTM.

One question (that can be addressed later), you said:

> Currently, we support plugins only for CJS. So, this change explicitly appends the `.cjs` extension.

Should we document this somewhere (e.g. in [Writing plugins](https://stylelint.io/developer-guide/plugins) or in the migration guide? I know this is "implicit" with the current state of Stylelint. but may be good to call out. I know it's not directly related to this PR (since it's only on test fixtures), but was just wondering.

**Comment by ybiquitous** [comment]
I don't think adding that to the migration guide is needed, but adding that to the existing document may be good.

But I believe it's best to support ESM plugins, not only adding it to the doc.

**Pull request merged**